base/regex.jl

```diff
@@ -221,6 +221,18 @@ findnext(r::Regex, s::AbstractString, idx::Integer) = throw(ArgumentError(
 ))
 findfirst(r::Regex, s::AbstractString) = findnext(r,s,firstindex(s))
 
+"""
+    SubstitutionString(substr)
+
+Stores the given string `substr` as a SubstitutionString, for use in regular expression
+substitutions. Most commonly constructed using the [`@s_str`](@ref) macro.
```

Comment:
Would be nice to show an example of how to use the macro, as the link between `@s_str` and `s"..."` isn't completely obvious.

Reply:
As I mentioned in the other comment, I'm leaning towards just merging this into the doc for `@s_str` and removing the doc here. That would also parallel what's done with `Regex` - `@r_str` has documentation, the type itself doesn't. The link in `replace`'s doc will then just have to be changed to "`r` is a SubstitutionString (constructed with [`@s_str`](@ref))", which would lead the user more directly to the actual construct they'd need to use in the code. Do you think that make sense?

Reply:
I'd rather keep this docstring, especially since you've written it. ;-) The fact that `Regex` isn't documented is rather something to fix: all public types and methods should have docstrings.

Reply:
I've added an example of using the `s"..."` macro and added `SubstitutionString` to exports.jl. Does this PR need anything else or is this good to go?